valhalla/mjolnir/ingest_transit.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include <algorithm>
+#include <list>
+#include <queue>
+#include <string>
+#include <thread>
+#include <unordered_set>
+#include <vector>
+
+#include <boost/property_tree/ptree.hpp>
+#include <valhalla/baldr/graphid.h>
+#include <valhalla/filesystem.h>
+#include <valhalla/midgard/pointll.h>
+#include <valhalla/proto/transit.pb.h>
+
+namespace valhalla {
+namespace mjolnir {
+
+/**
+ * @brief Processes the data from 'select_transit_tiles' into GraphTiles, adding the data into the
+ * valhalla graph.
+ *
+ * @param pt Property tree containing the hierarchy configuration
+ *             and other configuration needed to build transit.
+ * @param tiles queue of all tiles that have transit data (tiles that need processing)
+ * @param thread_count
+ * @return std::list<GraphId>
+ */
+std::list<baldr::GraphId> ingest_transit(const boost::property_tree::ptree& pt);
```

Comment:
why is this a list? if it can be a  vector or an unordered_set that is better

Reply:
It's because in line 625 -> the list is spliced and copied into the promises list.

Reply:
yeah i think we dont technically have to splice them together we could just move ranges out of vectors into another vector but honestly it doesnt matter. just ignore this comment :smile: